terraform/modules/department/50-aws-iam-roles.tf

```diff
@@ -10,7 +10,8 @@ data "aws_iam_policy_document" "sso_staging_user_policy" {
     data.aws_iam_policy_document.s3_department_access.json,
     data.aws_iam_policy_document.glue_access.json,
     data.aws_iam_policy_document.secrets_manager_read_only.json,
-    data.aws_iam_policy_document.redshift_department_read_access.json
+    data.aws_iam_policy_document.redshift_department_read_access.json,
+    data.aws_iam_policy_document.mwaa_department_web_server_access.json
```

Comment:
```
data "aws_iam_policy_document" "sso_staging_user_policy" {
  override_policy_documents = local.create_notebook ? [
    data.aws_iam_policy_document.s3_department_access.json,
    data.aws_iam_policy_document.glue_access.json,
    data.aws_iam_policy_document.secrets_manager_read_only.json,
    data.aws_iam_policy_document.redshift_department_read_access.json,
    data.aws_iam_policy_document.notebook_access[0].json
    ] : [
    data.aws_iam_policy_document.s3_department_access.json,
    data.aws_iam_policy_document.glue_access.json,
    data.aws_iam_policy_document.secrets_manager_read_only.json,
    data.aws_iam_policy_document.redshift_department_read_access.json,
    data.aws_iam_policy_document.mwaa_department_web_server_access.json
  ]
}
```

When create_notebook is true, do you need to add this line(mwaa permission)?

`data.aws_iam_policy_document.mwaa_department_web_server_access.json`

I have a quick review: create_notebook's default is null, which is derived from the variable notebook_instance.  It looks like you don't need to worry about this case when create_notebook is true.

Meanwhile, this only impacts the staging account, so I think it's safe to proceed at current stage.